src/storage/in_mem_storage_structure/in_mem_lists.cpp

```diff
@@ -18,18 +19,136 @@ PageElementCursor InMemLists::calcPageElementCursor(
     return cursor;
 }
 
-InMemLists::InMemLists(
-    std::string fName, LogicalType dataType, uint64_t numBytesForElement, uint64_t numNodes)
-    : fName{std::move(fName)}, dataType{std::move(dataType)}, numBytesForElement{
-                                                                  numBytesForElement} {
+InMemLists::InMemLists(std::string fName, LogicalType dataType, uint64_t numBytesForElement,
+    uint64_t numNodes, const common::CopyDescription* copyDescription, bool hasNullBytes)
+    : fName{std::move(fName)}, dataType{std::move(dataType)},
+      numBytesForElement{numBytesForElement}, copyDescription{copyDescription} {
     listsMetadataBuilder = make_unique<ListsMetadataBuilder>(this->fName);
     auto numChunks = StorageUtils::getListChunkIdx(numNodes);
     if (0 != (numNodes & (ListsMetadataConstants::LISTS_CHUNK_SIZE - 1))) {
         numChunks++;
     }
     listsMetadataBuilder->initChunkPageLists(numChunks);
-    inMemFile = make_unique<InMemFile>(this->fName, numBytesForElement,
-        this->dataType.getLogicalTypeID() != LogicalTypeID::INTERNAL_ID);
+    numElementsInAPage = PageUtils::getNumElementsInAPage(numBytesForElement, hasNullBytes);
+    inMemFile = make_unique<InMemFile>(this->fName, numBytesForElement, hasNullBytes);
+}
+
+void InMemLists::copyArrowArray(
+    arrow::Array* boundNodeOffsets, arrow::Array* posInRelLists, arrow::Array* array) {
+    switch (array->type_id()) {
+    case arrow::Type::BOOL: {
+        templateCopyArrayToRelLists<bool>(boundNodeOffsets, posInRelLists, array);
+    } break;
+    case arrow::Type::INT16: {
+        templateCopyArrayToRelLists<int16_t>(boundNodeOffsets, posInRelLists, array);
+    } break;
+    case arrow::Type::INT32: {
+        templateCopyArrayToRelLists<int32_t>(boundNodeOffsets, posInRelLists, array);
+    } break;
+    case arrow::Type::INT64: {
+        templateCopyArrayToRelLists<int64_t>(boundNodeOffsets, posInRelLists, array);
+    } break;
+    case arrow::Type::DOUBLE: {
+        templateCopyArrayToRelLists<double_t>(boundNodeOffsets, posInRelLists, array);
+    } break;
+    case arrow::Type::FLOAT: {
+        templateCopyArrayToRelLists<float_t>(boundNodeOffsets, posInRelLists, array);
+    } break;
+    case arrow::Type::DATE32: {
+        templateCopyArrayToRelLists<common::date_t>(boundNodeOffsets, posInRelLists, array);
+    } break;
+    case arrow::Type::TIMESTAMP: {
```

Comment:
I think we need to have a specialized function to handle timestamp dataType. The timestamp arrow array contains some metadata information including timezone and units.

Reply:
Left a todo for now.